search/indexing_api.py

```diff
@@ -109,12 +110,16 @@ def index_program_enrolled_users(program_enrollments, chunk_size=100):
     Returns:
         int: Number of indexed items
     """
-    return _index_chunks(
+    num_indexed = _index_chunks(
         (serialize_program_enrolled_user(program_enrollment) for program_enrollment in program_enrollments),
         USER_DOC_TYPE,
         chunk_size=chunk_size,
     )
 
+    for program_enrollment in program_enrollments:
+        send_automatic_emails(program_enrollment)
```

Comment:
All indexing is already done in asynchronous tasks. The above `_index_chunks` call can take a while, it's basically what `recreate_index` does